Add cache utility methods (#136)
velocidi/apso: docs/README.md

The `SerializableAWSCredentials` class provides a serializable container for AWS credentials, extending the `AWSCredentials` class (from AWS SDK for Java).

## Caching
The `apso-caching` module provides provides utilities for caching. 

To use it in an existing SBT project, add the following dependency to your `build.sbt`:

```scala
libraryDependencies += "com.velocidi" %% "apso-caching" % "@VERSION@"
```

> Apso provides utilities to simplify the caching of method calls, with [ScalaCache](https://cb372.github.io/scalacache/)'s `Guava` or `Caffeine`. These are provided in an implicit scope, to make its invocation simpler:

Reviewer: I would change that a little:
```suggestion
Apso provides utilities to simplify the caching of method calls, with [ScalaCache](https://cb372.github.io/scalacache/) 
and using either `Guava` or `Caffeine` as underlying cache implementations.

These utilities are provided as `.cached()` and `.cachedFn()` extension methods over all `FunctionN[]` types:
```

Author: Applied it on d258cbc8ba4b37f616816c0e7235134778fd07be.

```scala mdoc:reset
import scala.concurrent._
import scala.concurrent.duration._
import scala.concurrent.ExecutionContext.Implicits.global

import java.util.concurrent.atomic.AtomicInteger

import com.velocidi.apso.caching._

val x = new AtomicInteger(0)

val cachedFn = ((i: Int) => {
  val value = x.getAndAdd(i)
  value
}).cached(config.Cache.Caffeine(Some(3), None))

> cachedFn(2)
> cachedFn(2)

Reviewer: Nitpick:

```suggestion
cachedFn(1)
cachedFn(1)
x
```

Author: Applied on d258cbc8ba4b37f616816c0e7235134778fd07be.

val y = new AtomicInteger(0)

val cachedFutFn = ((i: Int) => Future {
  val value = y.getAndAdd(i)
  value
}).cachedF(config.Cache.Guava(Some(2), None))

> Await.result(cachedFutFn(3), Duration.Inf)
> Await.result(cachedFutFn(3), Duration.Inf)

Reviewer: Nitpick:

```suggestion
Await.result(cachedFutFn(1), Duration.Inf)
Await.result(cachedFutFn(1), Duration.Inf)
y
```

Author: Applied on d258cbc8ba4b37f616816c0e7235134778fd07be.
